### DISCUSSION TITLE
How to override base file?

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#7378] [Archive]
I have edit some .vue files in /app/src/components/v-form to add some custom function (a field can set value of another, based on certain condition).

I would like to know if could be possible to override these files instead of edit base project files, so that after update Directus my edit will not be overwritten.

Exist some override mechanism in Directus for base files?

Thank you!

**Comment by rijkvanzanten** [comment]
No, you can't override any of the source code files in the production release of Directus, as they're all bundled together into the final release

**Reply by ghost** [reply]
So the only way is fork from GitHub, make necessary edits and when necessary syncing the fork with upstream end eventually make a pull request to Directus/Directus if respond to 80/20 rule?

For example, my implementation resolve that proposed here: https://github.com/directus/directus/discussions/3583#discussioncomment-821578

How can I know if responds to 80/20 rule for submit pull request?

**Reply by rijkvanzanten** [reply]
> So the only way is fork from GitHub, make necessary edits and when necessary syncing the fork with upstream end eventually make a pull request to Directus/Directus if respond to 80/20 rule?

Yup! Or rely on extensions where possible. In this case, what you're asking for is a new feature in the app that allows interfaces to set values to other fields on the same form, which will have to be implemented in core. The best way to get that done is to write a detailed proposal with how you think it should work exactly, how you think to implement it, and what would be required to get it done. Then we can work on a PR and get it in 👍🏻

**Reply by ghost** [reply]
Perfect! Thank you for your support!
